cli/npm/installer.rs

```diff
@@ -76,17 +79,20 @@ impl PackageJsonDepsInstaller {
       .collect::<Vec<_>>();
     package_reqs.sort(); // deterministic resolution
 
-    inner
-      .npm_registry_api
-      .cache_in_parallel(
-        package_reqs.iter().map(|req| req.name.clone()).collect(),
-      )
-      .await?;
+    let mut req_with_infos =
+      FuturesOrdered::from_iter(package_reqs.into_iter().map(|req| {
+        let api = inner.npm_registry_api.clone();
+        async move {
+          let info = api.package_info(&req.name).await?;
+          Ok::<_, AnyError>((req, info))
+        }
+      }));
```

Comment:
Question: it appears we'll stop downloading package on first error encountered. Would it make sense to download as many as we can even if one of them errors?

Reply:
This is only the package's registry information which is quick to download. I'm not usre it's worth the complexity. For example, we'd have to consider what occurs if all of them error because of some network issue which could cause batches of them to timeout in series meaning it might take a long time to wait for all of them to error then finally surface the error.